src/webots/core/WbHttpReply.cpp

```diff
@@ -41,25 +41,30 @@ QByteArray WbHttpReply::forgeHTMLReply(const QString &htmlContent) {
   return reply;
 }
 
-QByteArray WbHttpReply::forgeImageReply(const QString &imageFileName) {
+QByteArray WbHttpReply::forgeFileReply(const QString &fileName) {
   QByteArray reply;
 
-  QFile imageFile(imageFileName);
-  if (!imageFile.open(QIODevice::ReadOnly))
+  QFile file(fileName);
+  if (!file.open(QIODevice::ReadOnly))
     return forge404Reply();
 
-  QByteArray imageData = imageFile.readAll();
-  int imageSize = imageData.length();
-  QFileInfo fi(imageFile);
-  QString imageExtension = fi.suffix().toLower();
-
+  const QByteArray data = file.readAll();
+  QFileInfo fi(file);
+  const QString extension = fi.suffix().toLower();
+  QString type;
+  if (extension == "png" || extension == "jpg" || extension == "jpeg")
+    type = QString("image/%1").arg(extension);
+  else if (extension == "html" || extension == "css")
+    type = QString("text/%1").arg(extension);
+  else if (extension == "js")
+    type = "application/javascript";
```

Comment:
I wouldn't add an assert here as we don't want to crash Webots in case a client robot window is requesting an unsupported file type, but I will definitely put a warning.